src/Solana/V0Message.h

```diff
@@ -0,0 +1,19 @@
+// Copyright © 2017-2023 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "Solana/LegacyMessage.h"
+#include "Solana/AddressLookupTable.h"
+
+namespace TW::Solana {
+
+struct V0Message {
```

Comment:
got a bit confused on this `struct`, isn't `V0Message` and `LegacyMessage` are same level? or it's better called `V0Transaction`?

Reply:
It's based on the official implementation of Solana in Rust: https://github.com/solana-labs/solana/tree/master/sdk/program/src/message/versions as you can see they have a version mod (versioned message) that can be either a V0Message either a LegacyMessage, V0Message is serialized differently, with a version prefix and address lookup table, but the message remain the same, that's why V0Message contains a LegacyMessage as a field to avoid code duplication. maybe @CriesofCarrots Can tell a bit more about why it's made this way in the rust codebase. Note that that's the same approach for VersionedTransaction: https://github.com/solana-labs/solana/tree/master/sdk/src/transaction/versioned

Reply:
In our rust implementation, [v0::Message doesn't wrap a LegacyMessage](https://github.com/solana-labs/solana/blob/48e2cdbc88a6d64d37e4e38756c00661c3086567/sdk/program/src/message/versions/v0/mod.rs#L55), but because all the LegacyMessage fields are duplicated, it seems fine to write it this way.
It's possible that v1+ messages may not support all those LegacyMessage fields, but it shouldn't be too difficult to deal with that in the future.

Reply:
Regarding `VersionedMessage`, it's written as an enum so that apis can handle both legacy and v0 messages -- and any future message versions -- without dynamic dispatch.

Reply:
> It's possible that v1+ messages may not support all those LegacyMessage fields, but it shouldn't be too difficult to deal with that in the future.

That's why I separated message into different classes, V0 is a composition of legacy + version number and table, and in the future indeed it should be easy to add V1Message with the new approach, thanks for the feedback!